### PR TITLE
Don't include unnecessary values in properties

### PR DESCRIPTION
**Pull request by nkzawa**
This PR removes unnecessary values in `vnode.properties` which derives from html attributes.

### CODE REVIEW

**Comment by azer** [comment]
@nkzawa thx for the PR! Could you split that to some more functions ?

**Comment by nkzawa** [comment]
Thanks for your quick response! Do you mean to split tests?

**Comment by azer** [comment]
@nkzawa smaller functions like the one createDataSet and style functions, to improve keep the main function small & more readable

**Comment by nkzawa** [comment]
Fixed and rebased. Could you check?

**Comment by nkzawa** [comment]
@azer Can you check this again? I will fix if there are any problems.

File: index.js [hunk 1]
```diff
@@ -64,25 +55,47 @@ function style (raw) {
...
-  var dataset;
-  var key;
-
-  for (key in props) {
-    if (key.slice(0, 5) == 'data-') {
-      dataset || (dataset = {});
-      dataset[camel(key.slice(5))] = props[key];
+function createProperties (attributes) {
+  var properties = {};
+
```
Comment (tnhu) [inline index.js:72]: In order to support custom attributes, add:

`var attrs = {};`

File: index.js [hunk 2]
```diff
@@ -64,25 +55,47 @@ function style (raw) {
...
+      properties.className = attributes['class'];
+      break;
+    case 'for':
+      properties.htmlFor = attributes['for'];
+      break;
+    default:
+      if (key.indexOf('data-') === 0) {
+        addDataSet(properties, key, attributes[key]);
+      } else {
+        properties[key] = attributes[key];
```
Comment (tnhu) [inline index.js:90]: In order to support custom attributes, replace this line with:

`attrs[key] = attributes[key];`

File: index.js [hunk 3]
```diff
@@ -64,25 +55,47 @@ function style (raw) {
...
+      }
     }
   }
```
Comment (tnhu) [inline index.js:94]: In order to support custom attributes, add:

`properties.attributes = attrs[key];`

**Comment by rangoo94** [comment]
What is status of this PR? It will simplify e.g. supporting lower IE versions when passing directly to `virtual-dom`, as then you can just push everything from `dataset` to `attributes`.

**Comment by azer** [comment]
Sorry for late reply. I can't maintain this repo anymore, any volunteers ? I'd like to transfer the ownership. 

@nkzawa @lamplightdev @Rangoo94

**Comment by rangoo94** [comment]
@azer, I can take care of this project, but I will have really small amount of time for contribution; mostly it will look like few hours of contribution per month and reviewing/merging PRs up to date.

**Comment by yoshuawuyts** [comment]
@azer I volunteer as tribute.

**Comment by azer** [comment]
thanks @Rangoo94! @yoshuawuyts I added you as collaborator. would you like to remove your fork so I can fully transfer the ownership to you?

**Comment by yoshuawuyts** [comment]
@azer done c:

**Comment by azer** [comment]
@yoshuawuyts is the new owner :) cheers

**Comment by yoshuawuyts** [comment]
I recommend folks use https://github.com/shama/bel instead of this package; we've been working hard on it for the past two years to make sure it has all the features you could want to write inline HTML. Hope it's good; closing for now!

**Pull request closed without merging**